Gems/Terrain/Code/Source/TerrainRenderer/TerrainDetailMaterialManager.cpp

```diff
@@ -697,17 +708,19 @@ namespace Terrain
                 UpdateDetailTexture(region.m_worldAabb, region.m_localAabb);
             }
 
-            m_dirtyDetailRegion = m_dirtyDetailRegion.GetClamped(untouchedRegion);
-
             if (m_dirtyDetailRegion.IsValid())
             {
-                ClipmapBounds::ClipmapBoundsRegionList updateRegions = m_detailMaterialIdBounds.TransformRegion(m_dirtyDetailRegion);
-                for (auto& region : updateRegions)
+                m_dirtyDetailRegion = m_dirtyDetailRegion.GetClamped(untouchedRegion);
+                if (m_dirtyDetailRegion.IsValid())
                 {
-                    UpdateDetailTexture(region.m_worldAabb, region.m_localAabb);
+                    ClipmapBounds::ClipmapBoundsRegionList updateRegions = m_detailMaterialIdBounds.TransformRegion(m_dirtyDetailRegion);
+                    for (auto& region : updateRegions)
```

Comment:
nit: this and the other similar loops in this file can be using const auto& instead of auto&